hredis/redigo/redigo.go

```diff
@@ -42,15 +44,61 @@ func WaitForAvailability(url string, d time.Duration, f WaitFunc) (bool, error)
 	}
 }
 
+// make this a var, so we can change this for testing.
+var redisDialURL = redis.DialURL
+
 // NewRedisPoolFromURL returns a new *redigo/redis.Pool configured for the supplied url
 // The url can include a password in the standard form and if so is used to AUTH against
 // the redis server
-func NewRedisPoolFromURL(url string) (*redis.Pool, error) {
+func NewRedisPoolFromURL(rawURL string, altPasses ...string) (*redis.Pool, error) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		findPassword sync.Once
+		password     string
+	)
+
+	// strip and save password
+	if pass, ok := u.User.Password(); ok {
+		password = pass
+	}
+
+	// DialURL will fail if wrong password is set. We want to create a successful connection
+	// with which we can try all of the possible passwords.
+	u.User = url.UserPassword("", "")
+	rawURL = u.String()
+
 	return &redis.Pool{
 		MaxIdle:     3,
 		IdleTimeout: 240 * time.Second,
 		Dial: func() (redis.Conn, error) {
-			return redis.DialURL(url)
+			c, err := redisDialURL(rawURL)
```

Comment:
When using redis.DialURL in running code won't this be too late to try other passwords? If I'm reading right, DialURL in turn calls Dial which [will then try and AUTH with the given password](https://github.com/gomodule/redigo/blob/39e2c31b7ca38b521ceb836620a269e62c895dc9/redis/conn.go#L229-L234), erroring if it doesn't work.

Reply:
Exactly why the password is cleared from the URL first. :) IF the password is blank, it won't issue `"AUTH"`

Reply:
Yargh, sorry. It's Friday! 🎉